test/FunctionalTests/Microsoft.OData.Client.Tests/ALinq/DataServiceContextMethodMockSampleTest.cs

```diff
@@ -0,0 +1,56 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="DataServiceContextMethodMockSampleTest.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+using System;
+using Moq;
+using Xunit;
+
+namespace Microsoft.OData.Client.Tests.ALinq
+{
+    /// <summary>
+    /// How to mock the DataServiceContext class methods using moq framework example.
+    /// </summary>
+
+    public class DataServiceContextMethodMockSampleTest
+    {
+        [Fact]
+        public void VerifyIfDataObjectMethodIsMockable()
+        {
+            string str = "user";
+            Mock<DataServiceContext> mockObject = new Mock<DataServiceContext>();
+            mockObject.Setup(x => x.AddObject(It.IsAny<string>(), It.IsAny<object>())).Callback(()=> { str = "Saved Successfully"; });
+            SaveUserDetails saveUserDetails = new SaveUserDetails() ;
+            string result = saveUserDetails.SaveUser(mockObject.Object);
+            Assert.Equal(str, result);
+        }
+    }
+
+    /// <summary>
+    /// A sample implementation class with methods calling the DataserviceContext.cs methods
+    /// </summary>
+    public class SaveUserDetails

```

Comment:
Class Names, variable names and property names ideally should be Nouns, only functions, actions and methods should be verbs.